### PR TITLE
Add Basic-Auth support

### PR DESCRIPTION
**Pull request by alex-schneider**

### CODE REVIEW

File: errors/http_status.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ func httpStatus(code Code) int {
 		return http.StatusUnauthorized
 	case AuthorizationFailed:
 		return http.StatusForbidden
+	case BasicAuthFailed:
```
Comment (malud) [inline errors/http_status.go:17]: This case should be merged with `AuthorizationRequired` case.

File: errors/template.go [hunk 2]
```diff
@@ -68,6 +68,11 @@ func (t *Template) ServeError(errCode Code) http.Handler {
 		rw.Header().Set(HeaderErrorCode, fmt.Sprintf("%d - %q", errCode, errCode))
 
 		status := httpStatus(errCode)
+
+		if status == http.StatusUnauthorized {
```
Comment (malud) [inline errors/template.go:72]: `WWW-Authenticate` header handling should find a place with basic auth context. The same http status code could occur for all other access control variants.

Comment (alex-schneider) [reply errors/template.go]: Done

File: errors/http_status.go (hunk 1 above)
Comment (alex-schneider) [reply errors/http_status.go]: Done

File: accesscontrol/basic_auth.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+type BasicAuth struct {
+	htFile htData
+	name   string
+	user   string
+	pass   string
+	realm  string
+}
+
+// NewBasicAuth creates a new AC-BasicAuth object
+func NewBasicAuth(name, user, pass, file, realm string) (*BasicAuth, error) {
```
Comment (johakoch) [inline accesscontrol/basic_auth.go:33]: `realm` is currently not used, but should go into the `WWW-Authenticate` response header.

**Review by malud — APPROVED** [review]

**Review by johakoch — APPROVED** [review]

**Pull request merged**